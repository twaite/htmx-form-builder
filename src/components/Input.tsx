import clsx from "clsx";

type Props = {
  inputClasses?: string;
  label?: string;
  name: string;
  error?: string;
  required?: boolean;
  type?: "text" | "email" | "password" | "textarea";
  value?: string;
};

export default function Input(props: Props) {
  const inputProps = {
    name: props.name,
    autofocus: props.error,
    class: clsx(
      "text-black border-2 dark:focus:border-emerald-500 p-2 rounded outline-none dark:bg-slate-200",
      props.inputClasses,
      {
        "border-red-400 focus:border-red-600 dark:focus:border-red-600":
          props.error,
      }
    ),
    required: props.required,
  };

  return (
    <Html.Fragment>
      {props.label && <label for={props.name}>{props.label}</label>}
      {props.type === "textarea" ? (
        <textarea {...inputProps}>{props.value}</textarea>
      ) : (
        <input type="text" {...inputProps} value={props.value} />
      )}
      {props.error && <p class="text-red-400">{props.error}</p>}
    </Html.Fragment>
  );
}
