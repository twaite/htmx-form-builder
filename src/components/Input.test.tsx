import { describe, test, expect } from "bun:test";
import Input from "./Input";

describe("Input", () => {
  test("should match snapshot", () => {
    expect(<Input name="test" />).toMatchSnapshot();
  });
});
