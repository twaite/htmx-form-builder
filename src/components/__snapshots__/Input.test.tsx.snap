// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `"<input type="text" name="test" class="text-black border-2 dark:focus:border-emerald-500 p-2 rounded outline-none dark:bg-slate-200"/>"`;
